deid/dicom/pixels/clean.py

```diff
@@ -108,10 +111,28 @@ def clean(self):
 
             # Load in dicom file, and image data
             dicom = read_file(self.dicom_file, force=True)
+            pixel_data = getattr(dicom, pixel_data_attribute)
+
+            # Get expected and actual length of the pixel data (bytes, expected does not include trailing null byte)
+            expected_length = get_expected_length(dicom)
+            actual_length = len(pixel_data)
+            padded_expected_length = expected_length + expected_length % 2
 
-            # We will set original image to image, cleaned to clean
-            self.original = dicom.pixel_array
-            self.cleaned = self.original.copy()
+            # If we have YBR_FULL_2, must be RGB to obtain pixel data
+            if (
+                actual_length > padded_expected_length
+                and dicom.PhotometricInterpretation != "RGB"
+                and fix_interpretation
+            ):
```

Comment:
Non-RGB PIs (like MONOCHROME1) can have excess padding as well, it's usually only a few bytes worth. That's why the pydicom numpy handler explicitly checks to see if the data is a third larger when YBR_FULL_422.

So a MONOCHROME1 dataset with excess padding here would end up changed to RGB, which is probably not desired behaviour

Reply:
okay, so how about just explicitly checking for `== "YBR_FULL_422"` - would that work?

Reply:
I would do:
```python
if ds.PhotometricInterpretation == "YBR_FULL_422" and fix_intepretation:
    full_length = expected_length / 2 * 3  # upsampled data is a third larger
    full_length += 1 if full_length % 2 else 0   # trailing padding byte if odd length
    if actual_length >= full_length:
        # warning, conversion, etc
```

Reply:
Also, I'd check to make sure the transfer syntax isn't compressed

Reply:
okay, I updated the above but moved the calculation of full length outside the loop so all 4 checks can be done at once. Does it look better now?
```python
# Get expected and actual length of the pixel data (bytes, expected does not include trailing null byte)
expected_length = get_expected_length(dicom)
actual_length = len(pixel_data)
padded_expected_length = expected_length + expected_length % 2
full_length = expected_length / 2 * 3  # upsampled data is a third larger
full_length += (
    1 if full_length % 2 else 0
)  # trailing padding byte if even length

# If we have YBR_FULL_2, must be RGB to obtain pixel data
if (
    not dicom.file_meta.TransferSyntaxUID.is_compressed
    and dicom.PhotometricInterpretation == "YBR_FULL_422"
    and fix_interpretation
    and actual_length >= full_length
):
    bot.warning(
        "Updating dicom.PhotometricInterpretation to RGB, set fix_interpretation to False to skip."
    )
    photometric_original = dicom.PhotometricInterpretation
    dicom.PhotometricInterpretation = "RGB"
    self.original = dicom.pixel_array
    dicom.PhotometricInterpretation = photometric_original
else:
    self.original = dicom.pixel_array
```

Reply:
Also @scaramallion can I add you as "coauthored by" to the final merge?

Reply:
Yep looks good.

>Also @scaramallion can I add you as "coauthored by" to the final merge?

Sure